src/per/adc.h

```diff
@@ -48,15 +48,17 @@ struct AdcChannelConfig
     \param mux_2 Third mux pin
     \param adc_pin &
     */
-    void InitMux(dsy_gpio_pin adc_pin,
-                 size_t       mux_channels,
-                 dsy_gpio_pin mux_0,
-                 dsy_gpio_pin mux_1 = {DSY_GPIOX, 0},
-                 dsy_gpio_pin mux_2 = {DSY_GPIOX, 0});
-
-    dsy_gpio pin_;                   /**< & */
-    dsy_gpio mux_pin_[MUX_SEL_LAST]; /**< & */
-    uint8_t  mux_channels_;          /**< & */
+    void InitMux(Pin    adc_pin,
+                 size_t mux_channels,
+                 Pin    mux_0,
+                 Pin    mux_1 = Pin(),
```

Comment:
We could also provide `Pin::invalid` like this
```cpp
struct Pin
{
    // ...
    static constexpr Pin invalid = {DSY_GPIOX, 0};
}
```
and write:
```cpp
void InitMux(Pin    adc_pin,
                 size_t mux_channels,
                 Pin    mux_0,
                 Pin    mux_1 = Pin::invalid,
                 Pin    mux_2 = Pin::invalid);
```

Reply:
Are you sure this is compatible with Pin being POD?
I get `a member of type "const daisy::Pin" cannot have an in-class initializer`
I'm sure it'll work if we have INVALID_PIN as a define, or maybe just not a member of Pin.

Reply:
Hmm, you're right, that's weird.
[This solution works though](https://godbolt.org/#g:!((g:!((g:!((g:!((h:codeEditor,i:(fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:20,positionColumn:2,positionLineNumber:20,selectionStartColumn:2,selectionStartLineNumber:20,startColumn:2,startLineNumber:20),source:'%23include+%3Cstdint.h%3E%0A%0Aenum+class+Port%0A%7B%0A++++DSY_GPIOA,%0A++++//+...%0A++++DSY_GPIOX%0A%7D%3B%0A%0Astruct+Pin%0A%7B%0A++++Port+port+%3D+Port::DSY_GPIOX%3B%0A++++uint8_t+pin+%3D+0%3B%0A++++static+constexpr+Pin+invalid()+%7B+return+%7B+Port::DSY_GPIOX,+0+%7D%3B+%7D%0A%7D%3B%0A%0Avoid+test()%0A%7B%0A++++Pin+p+%3D+Pin::invalid()%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:47.74396642182581,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g111,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'x86-64+gcc+11.1+(Editor+%231,+Compiler+%231)+C%2B%2B',t:'0')),k:52.256033578174176,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',m:63.31811263318112,n:'0',o:'',t:'0'),(g:!((h:output,i:(compiler:1,editor:1,fontScale:14,fontUsePx:'0',wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+11.1+(Compiler+%231)',t:'0')),header:(),l:'4',m:36.68188736681888,n:'0',o:'',s:0,t:'0')),l:'3',n:'0',o:'',t:'0')),version:4)